### PR TITLE
Improve SQL docs for creating and dropping repositories

### PR DESCRIPTION
**Pull request by norosa**
Spotted an opportunity to tidy up the SQL syntax docs for `CREATE REPOSITORY` and went for it.

Specifically:

- For `create-repository.rst`:

  - Improve intro and add cross-references

  - Add missing ref labels and tidy up nomenclature of ref labels throughout the whole doc

  - General copyedits for clarity, style, grammar, and so on

  - Tidied up and improved section/headings hierarchy and use of TOCs

  - Converted all params to use definition-style markup

        **Example**
          This is an example of the style used.

  - Improved use of admonitions and expanded/reworked explanations where necessary to improve clarity

  - Minor edits for the statement synopsis to improve readability (i.e., `repository_parameter` renamed `parameter_name`)

  - Standardized wording of duplicate parameter definitions

  - Standardized ordering of parameters

  - Added external links where helpful (e.g., to specific Hadoop, AWS, and Azure resources)

  - Corrected `readonly` parameter to `url` for `url` type repositories

- For `drop-repository.rst`:

  - Changes to match the above

- For other files:

  - Update ref labels to match changes above

### CODE REVIEW

**Comment by norosa** [comment]
@mkleen while rewording for clarity, I may have introduced errors. my recordings reflect my best understanding of what is being documented. in a few instances, I double-checked with the Elasticsearch docs or by looking at the source code directly. but there is a chance I have missed something subtle so I welcome close scrutiny

File: docs/sql/statements/create-repository.rst [hunk 1]
```diff
@@ -1,303 +1,442 @@
...
-(e.g. EC2).
 
+An ``s3`` repository stores snapshot on the `Amazon Simple Storage Service`_
+(Amazon S3).
+
+.. NOTE::
+
+    If you are using Amazon S3 in conjunction with `IAM roles`_, the
+    ``access_key`` and ``secret_key`` parameters must be left undefined. The
+    IAM role will provide the necessary credentials when required.
```
Comment (mkleen) [inline docs/sql/statements/create-repository.rst:295]: I think it is important to mention that when Amazon I AM Roles are used for authentication, that these roles must be provided by the underlying container where CrateDB is deployed.

**Review by mkleen — CHANGES_REQUESTED** [review]
Very good work, i left one point to follow up.

**Comment by norosa** [comment]
@mkleen thanks for the review! pushed a fixup. can you check? thx!

File: docs/sql/statements/create-repository.rst [hunk 2]
```diff
@@ -1,303 +1,445 @@
...
+
+.. NOTE::
+
+    If you are using Amazon S3 in conjunction with `IAM roles`_, the
+    ``access_key`` and ``secret_key`` parameters must be left undefined.
 
-.. rubric:: Parameters
+    Additionally, make sure to `attach the IAM to each EC2 instance`_ that will
+    run a CrateDB master node or data node. The attached IAM role will provide
+    the necessary credentials when required.
```
Comment (norosa) [inline docs/sql/statements/create-repository.rst:298]: @mkleen changed this

Comment (mkleen) [reply docs/sql/statements/create-repository.rst:298]: Perfect, thank you!

**Review by mkleen — APPROVED** [review]
👍

**Pull request merged**